pkg/agent/agent.go

```diff
@@ -183,12 +187,18 @@ func ParseDriverType(raw string) (StorageForDriver, error) {
 	}
 }
 
-func appOverviewFromRelease(r *release.Release) proxy.AppOverview {
-	return proxy.AppOverview{
-		ReleaseName: r.Name,
-		Version:     strconv.Itoa(r.Version),
-		Icon:        r.Chart.Metadata.Icon,
-		Namespace:   r.Namespace,
-		Status:      r.Info.Status.String(),
+func appOverviewFromRelease(r *release.Release) (proxy.AppOverview, error) {
+	releaseHelm2, err := helm3to2.Convert(*r)
```

Comment:
Converting the release to a helm 2 release seems (1) unnecessary and (2) the only reason for requiring an error in the signature of this function? That is, if `helm3to2.ConvertMaintainers(r.Chart.Metadata.Maintainers)` could be used below instead (ie. just making `compatableMaintainers` public), then we don't need to handle the error at all (nor convert the whole release).

Fine either way, just a thought.

Reply:
Well, we need the whole metadata (not just the maintainers) so the function to expose would be `compatibleMetadata` but yes, we could do that. I wanted to to keep the methods exposed by the compatibility package as minimum as possible but there is not a good reason for not exposing the metadata conversion.

Reply:
I for one see no problem with exporting `helm3to2` functions at will. They're all supposed to be pure and reusable anyway.

Reply:
Although one must keep in mind that some of the type signatures in `helm3to2` are essentially lies, since it's impossible in general to write a function from Helm 2 releases to Helm 3 releases.